### PR TITLE
allow richtext on papers + edit premade documents

### PR DESCRIPTION
**Pull request by onoira**
## About the PR

let players markup their fancy papers _and_ edit existing ones (character sheet, books, …). it is not possible to edit existing documents today. includes a guidebook entry ('Writing').

should be checked for regressions to https://github.com/space-wizards/space-station-14/pull/17786 (_i didn't see any_)

Known issues:

- [x] `KeybindTag` is still usable for some reason
- [x] `ColorTag` doesn't work on paper
- [x] ~~markup-only changes sometimes aren't saved (but you can temporarily change the text as a workaround)~~ (can't reliably reproduce)

Depends on https://github.com/space-wizards/RobustToolbox/pull/4187

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, ~~**or** this PR does not require an ingame showcase~~

image
image
image

**Changelog**

:cl:
- fix: Nanotrasen no longer uses disappearing ink for official books and documents. These can now be modified.
- add: New pen technology enables stylish ways of writing. See the guidebook for more info.

### CODE REVIEW

**Comment by mirrorcult** [comment]
test fail is erroneous

**Review by juliangiebel — CHANGES_REQUESTED** [review]
> caveats: (I don't consider any of these blockers)
>
> KeybindTag is still usable for some reason.
> ColorTag doesn't work on paper.
> markup-only changes sometimes aren't saved, but you can temporarily change the text as a workaround.

Those should be fixed before merging this PR imo.

The color not working might be because of the style that gets applied to paper `RichTextLabel`
```cs
  Element<RichTextLabel>().Class("PaperWrittenText")
      .Prop(Label.StylePropertyFont, notoSans12)
      .Prop(Control.StylePropertyModulateSelf, Color.FromHex("#111111")),
```

**Comment by moonheart08** [comment]
> > caveats: (I don't consider any of these blockers)
> > KeybindTag is still usable for some reason.
> > ColorTag doesn't work on paper.
> > markup-only changes sometimes aren't saved, but you can temporarily change the text as a workaround.
> 
> Those should be fixed before merging this PR imo.
> 
> The color not working might be because of the style that gets applied to paper `RichTextLabel`
> 
> ```cs
>   Element<RichTextLabel>().Class("PaperWrittenText")
>       .Prop(Label.StylePropertyFont, notoSans12)
>       .Prop(Control.StylePropertyModulateSelf, Color.FromHex("#111111")),
> ```

Agreeing with Julian, especially as more markup gets added for use in guides/etc. Last thing we need is for players to be able to insert UI elements like links when those are introduced.

**Review by juliangiebel — APPROVED** [review]
Looks good to me 👍

File: Content.Server/Paper/PaperSystem.cs [hunk 1]
```diff
@@ -134,10 +134,8 @@ private void OnInputTextMessage(EntityUid uid, PaperComponent paperComp, PaperIn
             if (string.IsNullOrEmpty(args.Text))
                 return;
 
-            var text = FormattedMessage.EscapeText(args.Text);
-
-            if (text.Length + paperComp.Content.Length <= paperComp.ContentSize)
-                paperComp.Content = text;
+            if (args.Text.Length + paperComp.Content.Length <= paperComp.ContentSize)
+                paperComp.Content = args.Text;
```
Comment (metalgearsloth) [inline Content.Server/Paper/PaperSystem.cs:138]: Server should be validating the text, no?

**Pull request merged**